toolbox/timecomparer/main.go

```diff
@@ -0,0 +1,532 @@
+			}
+
+			if realOutput.MatchString(fileSlice[start]) && userOutput.MatchString(fileSlice[start+1]) && sysOutput.MatchString(fileSlice[start+2]) {
+				if fileStack.Len() != 0 {
+					previousCommand := fileStack.Pop()
+					corespondingCommand := previousCommand.(string)
+					// Find each section of time in the output
+					realTime := extractTime(fileSlice[start])
+					userTime := extractTime(fileSlice[start+1])
+					sysTime := extractTime(fileSlice[start+2])
+					average := getAverageTime(realTime, userTime, sysTime)
+
+					rTime := runTime{
+						runTime: []runPair{
+							realTime, userTime, sysTime,
+						},
+						runPath: filePath,
+					}
+
+					empty := runCombination{}
+					if reflect.DeepEqual(commandToTime[corespondingCommand], empty) {
+						commandToTime[corespondingCommand] = runCombination{
+							totalTime: average,
+							runTimes:  []runTime{rTime},
+						}
+					} else {
+						rCombination := commandToTime[corespondingCommand]
+						rCombination.totalTime += average
+						rCombination.runTimes = append(rCombination.runTimes, rTime)
+						commandToTime[corespondingCommand] = rCombination
+					}
+				}
+			}
+		}
+	}
+	return commandToTime
+}
+
+// For each spliter section of file paths, read previously generated errors and warnings.
+// Process contents in new files and update the csv.
+func (tc *TimeComparer) findErrorForEachSection(ctx context.Context, gcsFilePaths []string, outputFileName string, startInd int, endInd int) {
+	readResult, commandToTime, err := readCSV(outputFileName)
+
+	// If csv file does not contain anything yet, initialize curErrors and curWarnings to be empty maps.
+	if err != nil {
+		commandToTime = map[string]runCombination{}
+		readResult = map[string][]string{}
+	}
+	filesToProcess := append([]string{}, gcsFilePaths[startInd:endInd]...)
+
+	// Get time/command and pull request paths from build log.
+	commandToTime = tc.findTimeCommands(ctx, filesToProcess, commandToTime)
+
+	sortedArray, timeMap := processCommand(commandToTime)
+
+	// Sort the time values
+	sort.Float64s(sortedArray)
+	// Write combined content to output csv file.
+	writeCSV(sortedArray, readResult, timeMap, outputFileName)
+}
+
+// Divide the list of file paths read in gcs storage to several spliters to avoid overflooding memory.
+func (tc *TimeComparer) divideToSections(ctx context.Context, spliter int, gcsFilePaths []string, outputFileName string) {
+	n := 1
+	for {
+		if n*spliter > len(gcsFilePaths) {
+			break
+		}
+		startInd := (n - 1) * spliter
+		endInd := n*spliter - 1
+		tc.findErrorForEachSection(ctx, gcsFilePaths, outputFileName, startInd, endInd)
+		n++
+	}
+	startInd := (n - 1) * spliter
+	endInd := len(gcsFilePaths)
+	tc.findErrorForEachSection(ctx, gcsFilePaths, outputFileName, startInd, endInd)
+
+}
+
+// Optional on whether or not to copy and share output file to gcs.
+func (tc *TimeComparer) CopyToGCS(ctx context.Context, source, bucketName, fileName string, public bool) {
```

Comment:
What other package needs to use `CopyToGCS` right now? If none please do not export it`